package.json

```diff
@@ -35,27 +36,58 @@
     "fix": "npm run fix:lint && npm run fix:prettier",
     "start": "npm run develop",
     "serve": "gatsby serve",
-    "test": "echo \"Write tests! -> https://gatsby.app/unit-testing\"",
-    "commit": "git-cz"
+    "commit": "git-cz",
+    "cypress:open": "cross-env TZ=UTC node_modules/.bin/cypress open",
+    "cypress:run": "cross-env TZ=UTC node_modules/.bin/cypress run",
+    "test": "cross-env NODE_ENV=test npm run test:unit && npm run test:e2e",
+    "test:unit": "nyc --silent --no-clean jest",
+    "test:e2e": "nyc --silent --no-clean start-server-and-test develop http://localhost:8000 cypress:run",
+    "pretest": "chmod +x scripts/**.* && cross-env ./scripts/pre-test.sh",
+    "posttest": "cross-env ./scripts/build_report.sh"
```

Comment:
This will always send reports even on `development`, we might want to add a check in the script to only send a report if the environment is `TESTING`

Reply:
Your point is we should only run `build_report` during `TESTING`?

Reply:
Yes, since we don't want to always be sending coverage reports in our development environment.
You can the script to the `CircleCI` configuration.

Reply:
Alright. I think I understand what you mean.
We can short-circuit the script if `NODE_ENV=PRODUCTION`.

Reply:
There's going to be only one environment when we need to sending coverage when `NODE_ENV === 'testing'`
Every other environment shouldn't be sending coverage.

Reply:
Alright. Done.